packages/core/src/modules/routing/protocol/pickup/v1/MessagePickupService.ts

```diff
@@ -0,0 +1,64 @@
+import type { InboundMessageContext } from '../../../../../agent/models/InboundMessageContext'
+import type { EncryptedMessage } from '../../../../../types'
+import type { BatchPickupMessage } from './messages'
+
+import { Dispatcher } from '../../../../../agent/Dispatcher'
+import { EventEmitter } from '../../../../../agent/EventEmitter'
+import { createOutboundMessage } from '../../../../../agent/helpers'
+import { InjectionSymbols } from '../../../../../constants'
+import { inject, injectable } from '../../../../../plugins'
+import { MessageRepository } from '../../../../../storage/MessageRepository'
+
+import { BatchHandler, BatchPickupHandler } from './handlers'
+import { BatchMessage, BatchMessageMessage } from './messages'
+
+@injectable()
+export class MessagePickupService {
```

Comment:
Should this have a prefix to avoid clashes? We've taken this approach for issue credential / present proof. This would mean we also need to update the messages, handlers, etc...

```suggestion
export class V1MessagePickupService {
```

Reply:
I think it should be prefixed. And in this case, protocol messages are named differently so I guess there will be no need to prefix all classes.

Reply:
Hmm probably it's good to start doing it everywhere as it is leading to issues over time? Not sure